crates/aiken/src/cmd/tx/simulate.rs

```diff
@@ -85,6 +97,71 @@ pub fn exec(
 
     let tx = MultiEraTx::decode_for_era(Era::Conway, &tx_bytes).into_diagnostic()?;
 
+    let mut overrides: HashMap<ScriptHash, PlutusScript> = HashMap::new();
+
+    if !script_overrides.is_empty() {
+        with_project(None, false, true, false, |p| {
+            eprintln!(
+                "{} script overrides",
+                "      Computing"
+                    .if_supports_color(Stderr, |s| s.purple())
+                    .if_supports_color(Stderr, |s| s.bold()),
+            );
+            let hash_overrides: HashMap<ScriptHash, ScriptHash> = script_overrides
+                .iter()
+                .try_fold::<_, _, Result<_, aiken_project::error::Error>>(
+                    HashMap::new(),
+                    |mut acc, script_override| {
+                        let mut parts = script_override.split(":");
+
+                        let from = parts
+                            .next()
+                            .ok_or(Error::ScriptOverrideArgumentParseError {
+                                value: script_override.clone(),
+                            })?;
+
+                        let to = parts
+                            .next()
+                            .ok_or(Error::ScriptOverrideArgumentParseError {
+                                value: script_override.clone(),
+                            })?;
+
+                        let from_hash = ScriptHash::from(
+                            hex::decode(from)
+                                .map_err(|_| Error::ScriptOverrideArgumentParseError {
+                                    value: script_override.clone(),
+                                })?
+                                .as_slice(),
+                        );
+
+                        let to_hash = ScriptHash::from(
+                            hex::decode(to)
+                                .map_err(|_| Error::ScriptOverrideArgumentParseError {
+                                    value: script_override.clone(),
+                                })?
+                                .as_slice(),
+                        );
+
+                        acc.insert(from_hash, to_hash);
+                        Ok(acc)
+                    },
+                )?;
+
+            let blueprint_path = p.blueprint_path(blueprint.as_deref());
+            let blueprint = p.blueprint(&blueprint_path)?;
+            overrides = blueprint.into();
+            for (from, to) in hash_overrides {
+                let to_script = overrides
+                    .get(&to)
+                    .ok_or(Error::ScriptOverrideNotFound { script_hash: to })?;
+
+                overrides.insert(from, to_script.clone());
```

Comment:
I don't think that's correct? The resulting `overrides` will now **also** contain the original hash -> script mapping from the blueprint. 

Perhaps, instead of constructing a `HashMap<ScriptHash, ScriptHash>` above, we could already construct the overrides. So instead of `acc.insert(from_hash, to_hash)`, we could do:

```rust
   overrides.insert(from_hash, blueprint.get(&to_hash));
```

Reply:
`from` is the original hash, and `overrides` is a `HashMap`, so we're "upserting" here, if you will. But you're right, I kind of brute forced this, which leads to multiple iterations. Can definitely make this more efficient.